src/components/[guild]/Requirements/components/GuildCheckout/MintGuildPin/Fuel/hooks/useFuelBaseAssetBalance.ts

```diff
@@ -0,0 +1,13 @@
+import { BaseAssetId } from "fuels"
+import useFuel from "hooks/useFuel"
+import useSWR from "swr"
+
+const useFuelBaseAssetBalance = () => {
+  const { wallet, address } = useFuel()
+
+  const getBalance = async () => wallet.getBalance(BaseAssetId)
+
+  return useSWR(!!wallet ? ["fuelBaseAssetBalance", address] : null, getBalance)
```

Comment:
That's a great idea, but tbh since we'll use this only in one components, I don't think it's a big deal, so I'll leave it this way for now (I haven't noticed any performance / re-fetch issues fortunately) ;)